src/rust/lib/flexer/src/group.rs

```diff
@@ -0,0 +1,256 @@
+use crate::automata::pattern::Pattern;
+use crate::automata::nfa::NFA;
+use crate::group::rule::Rule;
+
+use itertools::Itertools;
+
+pub mod rule;
+
+/// Group of a set of rules.
+struct Group {
+    /// ID of group.
+    pub id     : usize,
+    /// Name of group.
+    pub name   : String,
+    /// Parent of group.
+    pub parent : Option<Box<Group>>,
+    /// Set of rules. See Rule for more information.
+    pub rules  : Vec<Rule>,
+    /// Function that is called on exiting group.
+    pub finish : Box<dyn FnMut()>,
+}
+
+impl Default for Group {
+    fn default() -> Self {
+        Group {
+            name   : Default::default(),
+            id     : Default::default(),
+            parent : Default::default(),
+            rules  : Default::default(),
+            finish : Box::new(||{}),
+        }
+    }
+}
```

Comment:
This is ugly a little bit. I feel that `finish` should not be a field in group - maybe it should be passed as other structure or other parameter? Anyway, all other fields should be grouped into a struct and `Default1 should be derived.